src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -540,33 +522,20 @@ private void RetrieveGeometry(string nodeId, string sessionId)
         /// custom nodes won't be removed</param>
         private void ClearWorkspace(bool clearOnlyHome)
         {
-            dynamoModel.Home(null);
+            // switch to home workspace
+            SelectTabByGuid(Guid.Empty);
 
             if (!clearOnlyHome)
             {
-                var customNodeManager = dynamoModel.CustomNodeManager;
-                var searchModel = dynamoModel.SearchModel;
-                var nodeInfos = customNodeManager.NodeInfos;
-
-                foreach (var guid in nodeInfos.Keys)
+                var allCustomNodeGuids = dynamoModel.CustomNodeManager.NodeInfos.Keys.ToList();
+                foreach (var guid in allCustomNodeGuids)
                 {
-                    searchModel.RemoveNodeAndEmptyParentCategory(guid);
-
-                    var name = nodeInfos[guid].Name;
-                    dynamoModel.Workspaces.RemoveAll(elem =>
-                    {
-                        // To avoid deleting home workspace 
-                        // because of coincidence in the names
-                        return elem != dynamoModel.HomeSpace && elem.Name == name;
-                    });
-
-                    customNodeManager.LoadedCustomNodes.Remove(guid);
+                    dynamoModel.CustomNodeManager.Remove(guid);
                 }
-
-                nodeInfos.Clear();
             }
 
-            dynamoModel.Clear(null);
+            dynamoModel.ClearCurrentWorkspace();
+            (dynamoModel.CurrentWorkspace as HomeWorkspaceModel).DynamicRunEnabled = false;
```

Comment:
`DynamicRunEnabled` is true means running is executed after every change (adding a node, changing a node value, etc.) that is not correct behavior for Flood which sends `RunCancelCommand` only when it's necessary